com/course/zenkouv/src/Lecture3/Task1/Knight.java

```diff
@@ -0,0 +1,124 @@
+import java.util.stream.Collectors;
+
+public class Knight {
+
+    private final String name;
+    private final List<Outfit> outfitList;
+    private int outfitsCost;
+
+    public Knight(String name) {
+        this.name = name;
+        this.outfitList = new ArrayList<>();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<Outfit> getOutfitList() {
+        return outfitList;
+    }
+
+    public int getOutfitsCost() {
+        return this.outfitsCost;
+    }
+
+    public void addOutfitToOutfitLis(Outfit outfit) {
+        if (!availabilityOutfit(outfit)) {
+            outfitList.add(outfit);
+        }
+    }
+
+    private boolean availabilityOutfit(Outfit outfit) {
+        if (isWeapon(outfit)) {
+            for (Outfit outfitFromList : outfitList) {
+                if (outfitFromList instanceof Weapon) {
+                    System.out.printf("I have a %s and me don't need any weapon!\n",
+                            ((Weapon) outfitFromList).getType().toString().toLowerCase());
+                    return true;
+                }
+            }
+        } else {
+            for (Outfit outfitFromList : outfitList) {
+                if (outfitFromList instanceof Armor) {
+                    if (((Armor) outfit).getType() == ((Armor) outfitFromList).getType()) {
+                        System.out.printf("I have a %s and me don't need any %s!\n",
+                                ((Armor) outfitFromList).getType().toString().toLowerCase(),
+                                ((Armor) outfitFromList).getType().toString().toLowerCase());
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean isWeapon(Outfit outfit) {
+        return outfit instanceof Weapon;
+    }
+
+    public void calculateOutfitsCost() {
+        if (haveAnyOutfits()) {
+            outfitsCost = outfitList.stream().mapToInt(Outfit::getCost).sum();
+        }
+    }
+
+    private boolean haveAnyOutfits() {
+        if (this.outfitList.isEmpty()) {
+            System.out.println("I haven't any outfit!");
+            return false;
+        }
+        return true;
+    }
+
+    public void compareOutfitsByCost() {
+        outfitList.sort(new CostComparator());
+    }
+
+    public void compareOutfitsByWeight() {
+        outfitList.sort(new WeightComparator());
+    }
+
+    public void printOutfitsByMinToMaxCost(int minCost, int maxCost) {
+        System.out.println(filterOutfitByMinToMaxCost(minCost, maxCost));
+    }
+
+    private List<Outfit> filterOutfitByMinToMaxCost(int minCost, int maxCost) {
+        return outfitList.stream().filter(outfit -> outfit.getCost() >= minCost && outfit.getCost() <= maxCost).collect(Collectors.toList());
+    }
+
+    public void printOutfitCost() {
+        System.out.printf("All my outfits cost %d \n", outfitsCost);
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
you don't compare instances of this class. 
what for you override hascode and equals?

Reply:
i thought i will have more than one knight object.
Deleted.